File: Binary Trees/branch_sums.py
def branch_sums(root):
    sums = []
    calculate_sums(root, 0, sums)
    return sums

def calculate_sums(node, total, sums):
    if not node:
        return
    
    newSum = total + node.value
    if not node.left and not node.right:
        sums.append(newSum)
        return
    calculate_sums(node.left, newSum, sums)
    calculate_sums(node.right, newSum, sums)

File: Binary Trees/branchSums.js
class BinaryTree {
    constructor(value) {
      this.value = value;
      this.left = null;
      this.right = null;
    }
  }

const branchSums = (root) => {
    const sums = []
	calculateSums(root, 0, sums)
	return sums
}

const calculateSums = (node, total, sums) => {
	if (!node) return
	
	const newSum = total + node.value
	if (!node.left && !node.right) {
		sums.push(newSum)
		return
	}
	calculateSums(node.left, newSum, sums)
	calculateSums(node.right, newSum, sums)
}
}